docs/contributor/draft/rate-limit-design.md

```diff
@@ -0,0 +1,80 @@
+|--------------------------------------------------|-------------|-----------------------------------------------------------------|-------------------------------|
+| wokloadSelector                                  | object      | when skipped, it's applied for the whole namespace workloads, if empty and in istio-system ns, applies to all workloads in all namespaces | no                            |
+| workloadSelector.labels                          | object      | allows to select given workload by a label selector                   | no                            |
+| local                                   | object      | allows to describe local rate limit properties                  | yes                           |
+| local.bucket                            | object      | allows to set rate limit capacity through bucket concept        | yes                           |
+| local.bucket.max_tokens                 | int         | max tokens that can be gathered for the rate limit capacity     | yes                           |
+| local.bucket.tokens_per_fill            | int         | number of tokens added per fill cycle                           | yes                           |
+| local.bucket.fill_interval              | time        | interval on which tokens are filled                             | yes                           |
+
+## Usage example
+
+```yaml
+apiVersion: operator.kyma-project.io/v1alpha1
+kind: RateLimit
+metadata:
+  name: httpbin-local-rate-limit
+  namespace: default
+spec:
+  context: "gateway" | "sidecar" # translates to: PatchContext: GATEWAY | SIDECAR_INBOUND, should be on this level to not to mix sidecar level rate limit with gateway level rate limit. In such case, 2 separate resources should be created.
+  workloadSelector:
+    labels:
+      app: httpbin
+  local:
+    bucket:
+      max_tokens: 1000
+      tokens_per_fill: 1000
+      fill_interval: 1s
+
+---
+
+# eventually we could extend it later into such direction
+
+apiVersion: operator.kyma-project.io/v1alpha1
+kind: RateLimit
+metadata:
+  name: default
+  namespace: kyma-system
+spec:
+  context: "gateway" | "sidecar" # translates to: PatchContext: GATEWAY | SIDECAR_INBOUND
+  workloadSelector:
+    labels:
+      app: httpbin
+  local:
+    bucket:
+      max_tokens: 1000
+      tokens_per_fill: 1000
+      fill_interval: 1s
+    time:
+      unit: "second" 
+      requestsLimit: 30
+    additionalCriteria:
+      headerValue:
+        time:
+          unit: "second" 
+          requestsLimit: 30
+        bucket:
+          max_tokens: 1000
+          tokens_per_fill: 1000
+          fill_interval: 1s
+        headerName: "some-header"
+        headerValue: "some-value"
+        vhost: "some.v.host"
+        route: "/some/path"
+  global:
+    globalSpecificSettings:
+```
+
+Additional details:
```

Comment:
Do we need this, because it should rather be mentioned in the ADR.

Reply:
I'll move this to the ADR